test/new-e2e/processes/linux_test.go

```diff
@@ -0,0 +1,227 @@
+
+	// Setup System Probe Config
+	sysProbeCfg := probeConfig{
+		SysProbeCfg: SystemProbeConfig{
+			Enabled: true,
+			ProcCfg: ProcessConfig{
+				Enabled: true,
+			},
+		},
+		NetworkCfg: NetworkConfig{
+			Enabled: true,
+		},
+	}
+
+	sysProbeYaml, err := yaml.Marshal(&sysProbeCfg)
+	require.NoError(s.T(), err)
+	s.T().Logf("%s", sysProbeYaml)
+	_, err = clients.ExecuteCommand(s.ec2.sshClient, "sudo touch /etc/datadog-agent/system-probe.yaml")
+	require.NoError(s.T(), err)
+	_, err = clients.ExecuteCommand(s.ec2.sshClient, "sudo chmod o+rw /etc/datadog-agent/system-probe.yaml")
+	require.NoError(s.T(), err)
+
+	file, err = s.sftpClient.OpenFile("/etc/datadog-agent/system-probe.yaml", os.O_WRONLY|os.O_CREATE|os.O_TRUNC)
+	require.NoError(s.T(), err)
+	_, err = file.Write(sysProbeYaml)
+	require.NoError(s.T(), err)
+
+	// Restart Process Agent for new agentConfig
+	_, err = clients.ExecuteCommand(s.ec2.sshClient, "sudo service datadog-agent restart")
+	require.NoError(s.T(), err)
+
+	s.waitForProcessAgent()
+
+	// Start containers
+	stdout, err = clients.ExecuteCommand(s.ec2.sshClient,
+		"docker run --name spring-hello -d springio/gs-spring-boot-docker")
+	s.T().Log(stdout)
+	assert.NoError(s.T(), err)
+
+	stdout, err = clients.ExecuteCommand(s.ec2.sshClient, "docker run --name sleepy  -d busybox sleep 9999")
+	s.T().Log(stdout)
+	assert.NoError(s.T(), err)
+
+	// Check processes are
+	ctx := datadog.NewDefaultContext(context.Background())
+	assert.Eventually(s.T(), func() bool {
+		resp, _, err := s.procAPIClient.ListProcesses(ctx, *datadogV2.NewListProcessesOptionalParameters().
+			WithSearch("java").
+			WithTags(fmt.Sprintf("%s,container_name:spring-hello", hostTag)).
+			WithPageLimit(2),
+		)
+		if err != nil {
+			s.T().Logf("Error when calling `ProcessesApi.ListProcesses`go: %v\n", err)
+			return false
+		}
+		if procs, ok := resp.GetDataOk(); ok && procs != nil {
+			return len(*procs) == 1
+		}
+		return false
+	}, 2*time.Minute, 1*time.Second)
+
+	assert.Eventually(s.T(), func() bool {
+		resp, _, err := s.procAPIClient.ListProcesses(ctx, *datadogV2.NewListProcessesOptionalParameters().
+			WithTags(fmt.Sprintf("%s,container_name:sleepy", hostTag)).
+			WithPageLimit(2),
+		)
+		if err != nil {
+			s.T().Logf("Error when calling `ProcessesApi.ListProcesses`: %v\n", err)
+			return false
+		}
+		if procs, ok := resp.GetDataOk(); ok && procs != nil {
+			return len(*procs) == 1
+		}
+		return false
+	}, 2*time.Minute, 1*time.Second)
+}
+
+func (s *LinuxTestSuite) waitForProcessAgent() {
+	var stdout string
+	require.Eventuallyf(s.T(), func() bool {
+		// Get process log
+		stdout, _ = clients.ExecuteCommand(s.ec2.sshClient,
+			"grep \"Starting process-agent for\" /var/log/datadog/process-agent.log")
+		return strings.Contains(stdout, "Starting process-agent for")
+	}, 2*time.Minute, 500*time.Millisecond, "process-agent is not running")
+
+	s.T().Log("process-agent is running")
+}
+
+type agentConfig struct {
```

Comment:
nit: It might be clearer if you combined this into a single struct like:

```go
type agentConfig struct {
	APIKey      string        `yaml:"api_key"`
	Hostname    string        `yaml:"hostname"`
	ProcessCfg  struct {
           ProcessCollection struct {
               Enabled bool `yaml:enabled,omitempty"`
           } `yaml:process_collection,omitempty"`
         } `yaml:"process_config,omitempty"`
	LogsEnabled bool          `yaml:"logs_enabled"`
}
```